src/cloudai/workloads/vllm/slurm_command_gen_strategy.py

```
@@ -42,42 +45,259 @@ def workload_name(self) -> str:
 
+    @staticmethod
+    def _with_ray_backend(command: list[str], enabled: bool) -> list[str]:
+        if not enabled or "--distributed-executor-backend" in command:
+            return command
+        return [*command, "--distributed-executor-backend", "ray"]
+
+    def _needs_ray(self, role: str) -> bool:
+        return self.role_node_count(role) > 1
+
+    @staticmethod
+    def _format_ray_value(value: Any) -> str:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)